docs/src/gravity_wave.md

```diff
@@ -0,0 +1,46 @@
+* The gravity wave propagates vertically and conservatively to the breaking level and deposits all momentum flux into that level, as opposed to the method using saturation profile described in [lindzen1981](@cite).
+* The wave breaking criterion is derived in a hydrostatic, non-rotating frame. Including nonhydrostatic and rotation is proved to have negligible impacts.
+* The gravity wave is intermittent and the intermittency is computed as the ratio of the total long-term average source to the integral of the source momentum flux spectrum.
+
+### Spetrum of the momentum flux sources
+The source spectrum with respect to phase speed is prescribed in Eq. (17) in [alexander1999](@cite). We adapt the equation so that the spectrum writes as a combination of a wide and a narrow band as
+```math
+B_0(c) = \frac{F_{S0}(c)}{\rho_0} = sgn(c-u_0) \left( Bm\_w \exp\left[ -\left( \frac{c-c_0}{c_{w\_w}} \right) \ln{2} \right] + Bm\_n \exp\left[ -\left( \frac{c-c_0}{c_{w\_n}} \right) \ln{2} \right] \right)
+```
+where the subscript ``0`` denotes values at the source level. ``c_0`` is the phase speed with the maximum flux magnitude ``Bm``. ``c_w`` is the half-width at half-maximum of the Gaussian.  ``\_w`` and ``\_n`` represent the wide and narrow bands of the spectra.
+
+### Upward propagation and wave breaking
+Waves that are reflected will be removed from the spectrum. A wave that breaks at a level above the source will deposit all its momentum flux into that level and be removed from the spectrum.
+
+The reflection frequency is defined as
+```math
+\omega_r(z) = (\frac{N(z)^2 k^2}{k^2+\alpha^2})^{1/2}
+```	
+where ``N(z)`` is the buoyancy frequency, ``k`` is the horizontal wavenumber that corresponds to a wavelength of 300 km, ``\alpha = 1/H`` where $H$ is the scale height. ``\omega_r(z)`` is used to determine for each monochromatic wave in the spectrum, whether it will be reflected at height ``z``.
+
+The instability condition is defined as
+```math
+Q(z,c) = \frac{\overline{\rho_0}}{\rho(z)} \frac{2N(z)B_0(c)}{k[c-u(z)]^3}
+```
+``Q(z,c)`` is used to determine whether the monochromatic wave of phase speed ``c`` gets unstable at height ``z``.
+
+* At the source level
+  - if ``|\omega|=k|c-u_0| \geq \omega_r``, this wave would have undergone internal reflection somewhere below and is removed from the spectrum;
+  - if ``Q(z_0, c) \geq 1``, it is also removed because it is not stable at the source level.
+
+* At the levels above ``(z_n>z_0)``, ``|\omega(z_n)|=k|c-u(z_n)| \geq \omega_r(z_n)`` is removed from the spectrum. In the remaining speed, ``Q(z_n,c) \geq 1`` are breaking between level ``z_{n-1}`` and ``z_n``, and this portion of momentum flux is all deposited between ``z_{n-1}`` and ``z_n``, which yields
+  ```math
+  X(z_{n-1/2}) = \frac{\epsilon}{\rho(z_{n-1/2)}}\Sigma_j (F_0)j.
+  ```
+  And we get 
+  ```math
+  X(z_{n-1}) = 0.5*\left[X(z_{n-3/2}) +X(z_{n-1/2}) \right].
+  ```
+
```

Comment:
Maybe add a sentence on how the tendency of u and v is calculated from the momentum flux X (Is it X or \chi)?